src/sources/file.rs

```diff
@@ -632,12 +658,16 @@ fn create_event(
     // Add source type
     event.insert(log_schema().source_type_key(), Bytes::from("file"));
 
-    if let Some(file_key) = &file_key {
+    if let Some(offset_key) = &meta.offset_key {
+        event.insert(offset_key.as_str(), offset.to_string());
```

Comment:
Giving a final look over this, I realise the offset almost certainly should not be cast to a string. Any objections to changing to the following before we merge? 
```suggestion
        event.insert(offset_key.as_str(), offset);
```

Reply:
In the interest of getting this merged sooner, I went ahead and made the change. This now results in field being serialised to json correctly:
```
{... "offset":248692, ...}
```
instead of
```
{... "offset":"248692", ...}
```